### DISCUSSION TITLE
Apple private relay

### DISCUSSION THREAD

**Discussion by ptman** [plausible/analytics#1347] [Anything Else]
According to https://rachelbythebay.com/w/2021/09/20/hop/ Apple private relay changes the source ip address every few seconds. How does this affect tracking?

**Comment by ukutaht** [comment]
This will of course affect Plausible tracking negatively since we rely on the IP address for unique user tracking and geolocation. Since this is an opt-in feature and a part of a paid iCloud subscription, it's likely to be enabled on a small amount of devices, though.

**Comment by ptman** [comment]
This got me thinking about ipv6 privacy extensions (temp addresses). How often do they usually change? Seems like once a day is the default on Linux. But what about other OSes?

**Reply by ukutaht** [reply]
Good question, I'll have to look into it.

Another question I don't know the answer to: do the privacy extensions change just the last byte or more? We're testing a version of Plausible that masks the last byte of every IP address. Works pretty well and it could solve this issue.

In general, clearly there are limitations with our tracking methods just like there are limitations with cookie-based tracking (disabling cookies, automatically clearing, etc). We take what the web standards give us and try to make the most of it. It will never be perfect.

Not to say we're not looking for ways to improve :) If you have suggestions shoot them my way

**Reply by ptman** [reply]
Typically the 128bit IPv6 address is split into 64bit network/prefix and 64bit local part. The local part changes in privacy extensions. And the change is supposed to be cryptographically random in order to provide good privacy.

**Reply by ukutaht** [reply]
Yeah OK, that will mean every time it changes we will track them as a new visitor
